worlds/cvcotm/client.py

```diff
@@ -0,0 +1,551 @@
+            max_hearts = int.from_bytes(health_stats_array[16:], "little")
+
+            # If there's no textbox already queued, the delay timer is 0, we are not in a cutscene, and Nathan's current
+            # state value is not 0x34 (using a save room), it should be safe to inject a textbox message.
+            ok_to_inject = not queued_textbox and not delay_timer and not cutscene \
+                and nathan_state != NATHAN_STATE_SAVING
+
+            # Make sure we are in the Gameplay or Credits states before detecting sent locations.
+            # If we are in any other state, such as the Game Over state, reset the textbox buffers back to 0 so that we
+            # don't receive the most recent item upon loading back in.
+            #
+            # If the intro cutscene floor broken flag is not set, then assume we are in the demo; at no point during
+            # regular gameplay will this flag not be set.
+            if game_state not in [GAME_STATE_GAMEPLAY, GAME_STATE_CREDITS] or not event_flags_array[6] & 0x02:
+                self.currently_dead = False
+                await bizhawk.write(ctx.bizhawk_ctx, [(QUEUED_TEXTBOX_1_ADDRESS, [0 for _ in range(12)], "EWRAM")])
+                return
+
+            # Enable DeathLink if it's in our slot_data.
+            if "DeathLink" not in ctx.tags and ctx.slot_data["death_link"]:
+                await ctx.update_death_link(True)
+
+            # Send a DeathLink if we died on our own independently of receiving another one.
+            if "DeathLink" in ctx.tags and hp == 0 and not self.currently_dead:
+                self.currently_dead = True
+
+                # Check if we are in Dracula II's arena. The game considers this part of the Sealed Room area,
+                # which I don't think makes sense to be player-facing like this.
+                if area == AREA_SEALED_ROOM and room == 2:
+                    area_of_death = "Dracula's realm"
+                # If we aren't in Dracula II's arena, then take the name of whatever area the player is currently in.
+                else:
+                    area_of_death = DEATHLINK_AREA_NAMES[area]
+
+                await ctx.send_death(f"{ctx.player_names[ctx.slot]} perished in {area_of_death}. Dracula has won!")
+
+            # Scout all Locations and get our Set events.
+            if ctx.locations_info == {}:
+                await ctx.send_msgs([{
+                    "cmd": "LocationScouts",
+                    "locations": [code for name, code in get_location_names_to_ids().items()
+                                  if code in ctx.server_locations],
+                    "create_as_hint": 0
+                }])
+                await ctx.send_msgs([{
+                    "cmd": "Get",
+                    "keys": [f"castlevania_cotm_events_{ctx.team}_{ctx.slot}"]
+                }])
+                # Some other parts of this need the scouted Location info and Set events, so return now.
+                return
```

Comment:
Changed so that it will now only send those packets once on initial connection, and a bool is used to track whether or not they have been sent. The client will also now skip features that require LocationScouts if it doesn't have them, including the in-game outgoing item textbox messaging system.